.github/workflows/cd.yml

```diff
@@ -0,0 +1,138 @@
+name: cd
+
+on:
+    push:
+        tags:
+        -   '[0-9]+.[0-9]+.[0-9]+*'
+
+jobs:
+
+    validate-release-tag:
+
+        if: github.repository == 'circus-test/circus'
+        runs-on: ubuntu-20.04
+
+        steps:
+        -   name: Checkout source
+            uses: actions/checkout@v2
+
+        -   name: Set up Python 3.8
+            uses: actions/setup-python@v2
+            with:
+                python-version: '3.8'
+
+        -   name: Validate the tag version against the package version
+            run: python .github/workflows/validate_release_tag.py $GITHUB_REF
+
+    ubuntu:
+
+        needs: [validate-release-tag]
+        runs-on: ubuntu-20.04
+
+        strategy:
+            fail-fast: false
+            matrix:
+                python-version: ['3.7', '3.8', '3.9', '3.10', 'pypy-3.7', 'pypy-3.8']
+                include:
+                -   python-version: '3.7'
+                    tox-env: py37
+                -   python-version: '3.8'
+                    tox-env: py38
+                -   python-version: '3.9'
+                    tox-env: py39
+                -   python-version: '3.10'
+                    tox-env: py310
+                -   python-version: pypy-3.7
+                    tox-env: pypy37
+                -   python-version: pypy-3.8
+                    tox-env: pypy38
+
+        steps:
+        -   uses: actions/checkout@v2
+
+        -   name: Cache Python dependencies
+            uses: actions/cache@v1
+            with:
+                path: ~/.cache/pip
+                key: pip-pre-commit-${{ hashFiles('**/setup.json') }}
+                restore-keys:
+                    pip-pre-commit-
+
+        -   name: Set up Python
+            uses: actions/setup-python@v2
+            with:
+                python-version: '3.8'
+
+        -   name: Install system dependencies
+            run: |
+                sudo rm -f /etc/apt/sources.list.d/dotnetdev.list /etc/apt/sources.list.d/microsoft-prod.list
+                sudo apt update
+                sudo apt install libev-dev libevent-dev
+                sudo apt install gcc make libffi-dev pkg-config zlib1g-dev libbz2-dev libsqlite3-dev libncurses5-dev libexpat1-dev libssl-dev libgdbm-dev tk-dev libgc-dev python-cffi liblzma-dev libncursesw5-dev
+                sudo ldconfig
+
+        -   name: Install test dependencies
+            run: pip install tox coveralls
+
+        -   name: Run test suite
+            run: tox -v -e $TOX_ENV
+            env:
+                TOX_ENV: ${{ matrix.tox-env }}
```

Comment:
Do we really have to re-launch all the tests? We usually tag a commit on master, for which the CI has already passed.

Reply:
We don't have to, it is just an extra safety that is usually applied in CD workflows. If someone merges something directly into master that breaks, it is better to catch it rather than pushing a broken build to PyPI. You cannot undo that. You will have to yank the release and create a new version. The CD workflow should not run that often, so running the tests one more time won't be that costly or time consuming

Reply:
My concern is mostly about the fact that we have to duplicate the pipeline which was defined in `ci.yml` :) . Also, unfortunately I did have to create releases with a red CI, because our CI is often unstable :( .

Reply:
You are both right in a sense that it is good to have full CI build before release and that circus' test suit is not stable. I think making sure that you can run publish stage manually is a good option if you know that this particular test can sometimes fail and it is ok.

I am not that familiar with GitHub Actions, so @sphuber can you confirm that it is possible to run publish stage manually even if tests are failing?

Reply:
Hi guys, appreciate the feedback. I think there is little risk. If the CI is just unstable and can fail occassionally, that should be no problem, because GithubActions allow you to rerun failed workflows with the click of a button. On top of that, even when that fails, nothing stops us from building locally and pushing manually like we used to.